### PR TITLE
Add Foreman 3.0 and Katello 4.2 release notes

### PR DESCRIPTION
**Pull request by ekohl**
This adds the release notes for Foreman 3.0.0 and Katello 4.2.0. It currently uses a script that I should submit somewhere. There's still a few issues that need to be fixed (markdown syntax in headline features), but I'm interested in the thoughts on the general layout.

I've chosen to have a structure where Foreman 3.0 and 4.2 each have their own main release notes section. Then there's a Foreman 3.0.0 section. Ideally, we've have a section Katello 4.2.0 above that. Then once Foreman 3.0.1 is release that gets shown. So it'll be:

1. Foreman 3.0 Release Notes
2. Katello 4.2 Release Notes
3. Foreman 3.0.1
4. Katello 4.2.0
5. Foreman 3.0.0
Appendix A: Foreman Contributors
Appendix B: Katello Contributors

Note that this is a PR to the 3.0 branch.

### CODE REVIEW

**Comment by lzap** [comment]
I don't like this chapter structure, it does not make sense:

image

First, we have Release Notes with features and warnings in two chapters, then there are chapters named Foreman and Katello. Why this is separate? This should be consistent, something like (just a proposal it can be different):

```
1. Foreman Release Notes

1.1 Foreman nightly release notes
1.1.1 Headline Features
1.1.2 Upgrade Warnings
1.1.3 Changes

1.2 Foreman 3.0 release notes
1.2.1 Headline Features
1.2.2 Upgrade Warnings
1.2.3 Changes

2. Katello Release Notes

2.1 Katello nightly release notes
2.1.1 Headline Features
2.1.2 Upgrade Warnings
2.1.3 Changes

2.2 Katello 4.2 release notes
2.2.1 Headline Features
2.2.2 Upgrade Warnings
2.2.3 Changes
```

This is a structure from nightly branch, for release branches there would be of course no "nightly" section - that is relevant only for master branch where we would collect features we are adding during development cycle.

File: guides/doc-Release_notes/topics/foreman.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[id="foreman-release-notes"]
+= Foreman {ProjectVersion} Release Notes
+
+[id="foreman-headline-features"]
+== Headline Features
+
+=== Puppet ENC functionality extracted to foreman_puppet plugin
+
+Foreman without the plugin no longer includes Puppet environments, config groups, Puppet classes and smart class parameters.
+It is still possible to assign Puppet environment to a host using the `puppet_environment` parameter. This is also the case for setting Puppet server and Puppet CA server for provisioning, which can be defined using the `puppet_server` and `puppet_ca_server`. A rake task will be provided to allow automatically migrating these values to parameters for users who do not wish to continue using the plugin.
```
Comment (maximiliankolb) [inline guides/doc-Release_notes/topics/foreman.adoc:10]:
```suggestion
It is still possible to assign Puppet environment to a host using the `puppet_environment` parameter.
This is also the case for setting Puppet server and Puppet CA server for provisioning, which can be defined using the `puppet_server` and `puppet_ca_server`.
A rake task will be provided to allow automatically migrating these values to parameters for users who do not wish to continue using the plugin.
```

**Review by maximiliankolb — APPROVED** [review]
Some suggestions, but the rest LGTM.

**Comment by ekohl** [comment]
Note to self: this needs to be updated with the latest changes in theforeman.org repo.

**Comment by lzap** [comment]
So this looks okay to me, at least it is a good starting point.

Maybe the subchapter "Foreman 3.0.0" could be named "Foreman 3.0.0 Changes" instead? Feels bare.

Other than that, please rebase and we are good to go. If you can create a template in master and drop some instructions somewhere (release engineering notes) that would be great.

**Comment by ekohl** [comment]
> Maybe the subchapter "Foreman 3.0.0" could be named "Foreman 3.0.0 Changes" instead? Feels bare.

I felt the same, but couldn't find a good phrasing. Yours sounds good.

**Review by lzap — APPROVED** [review]
Please update title and I am merging, thanks.

**Comment by ekohl** [comment]
This does not contain the latest changes. I need to get back to that so I've marked it as a draft for now. To prevent accidental merges.

**Comment by lzap** [comment]
If we could land this before 3.1 branching that would be awesome, thanks! Maybe @maximiliankolb could help since @ekohl is busy with getting the bits out.

**Comment by ekohl** [comment]
I think this is ready now.

**Review by maximiliankolb — COMMENTED** [review]
I am having a bit of a hard time reviewing this. I am unsure if we really want to build all possible variants. For example, there's release notes for orcharhino using saying "Foreman" instead of `{Project}`; there's something about ubuntu in satellite; and for satellite, it say's "Foreman 6.9" in the headline which is completely wrong.

Foreman-el looks good, though 👍

File: guides/doc-Release_notes/master.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
+include::common/attributes.adoc[]
+
+:numbered:
+:toc:
+:toc-placement: left
+
+ifndef::katello[]
+include::topics/foreman.adoc[]
```
[the remark this answers is not in the document]
Comment (ekohl) [reply guides/doc-Release_notes/master.adoc:8]: Can we consider this resolved due to https://github.com/theforeman/foreman-documentation/pull/698#discussion_r747503244 or is this a different discussion?

**Comment by ekohl** [comment]
> I am having a bit of a hard time reviewing this. I am unsure if we really want to build all possible variants. For example, there's release notes for orcharhino using saying "Foreman" instead of `{Project}`; there's something about ubuntu in satellite; and for satellite, it say's "Foreman 6.9" in the headline which is completely wrong.

I hadn't considered downstream in this since Satellite has its own release notes. Can we consider this an upstream-only document for now? If Orcharhino wants to use this a future PR would be welcome.

File: guides/doc-Release_notes/master.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/doc-Release_notes/master.adoc:8]: Ok, works for me!

**Comment by maximiliankolb** [comment]
> > I am having a bit of a hard time reviewing this. I am unsure if we really want to build all possible variants. For example, there's release notes for orcharhino using saying "Foreman" instead of `{Project}`; there's something about ubuntu in satellite; and for satellite, it say's "Foreman 6.9" in the headline which is completely wrong.
> 
> I hadn't considered downstream in this since Satellite has its own release notes. Can we consider this an upstream-only document for now? If Orcharhino wants to use this a future PR would be welcome.

OK, great, I did not know that.

In this case, I think we're good to go. Is #698 a prerequisite?

**Comment by ekohl** [comment]
> Is #698 a prerequisite?

Sort of. Technically this is against the 3.0 branch and the other to master. However, this PR includes a cherry pick of that PR. I should update this PR if the commit ID changes (as it happens when `Rebase and merge` is used). So if you think this is good to go, please merge that and I'll update this.

**Comment by maximiliankolb** [comment]
Merged https://github.com/theforeman/foreman-documentation/pull/698 to master; so you can cherry-pick the "proper" commit.

**Comment by ekohl** [comment]
Updated

**Review by maximiliankolb — APPROVED** [review]
LGTM. Can I merge this @ekohl ?

**Comment by ekohl** [comment]
Please do.

**Comment by lzap** [comment]
Was this supposed to be made available to readers? There is no link from any page, I also don't see this online:

https://docs.theforeman.org/Release_notes/

**Comment by ekohl** [comment]
The correct URLs are:
* https://docs.theforeman.org/3.0/Release_notes/index-foreman-el.html
* https://docs.theforeman.org/3.0/Release_notes/index-foreman-deb.html
* https://docs.theforeman.org/3.0/Release_notes/index-katello.html

**Pull request merged**